lib/elixir/src/elixir_module.erl

```diff
@@ -121,6 +122,32 @@ compile(Line, Module, Block, Vars, E) ->
     elixir_code_server:call({undefmodule, Ref})
   end.
 
+validate_compile_opts(Opts, Defs, Unreachable, File, Line) ->
+  lists:flatmap(fun (Opt) -> validate_compile_opt(Opt, Defs, Unreachable, File, Line) end, Opts).
+
+%% TODO: Make this an error and skip parse transform processing on 2.0.
+validate_compile_opt({parse_transform, Module} = Opt, _Defs, _Unreachable, File, Line) ->
+  elixir_errors:form_warn([{line, Line}], File, ?MODULE, {parse_transform, Module}),
+  [Opt];
+validate_compile_opt({inline, Inlines}, Defs, Unreachable, File, Line) ->
+  try [Inline || Inline <- Inlines, validate_inline(Inline, Defs, Unreachable)] of
+    [] -> [];
+    FilteredInlines -> [{inline, FilteredInlines}]
+  catch
+    {bad_inline, _} = Error ->
+      elixir_errors:form_error([{line, Line}], File, ?MODULE, Error)
+  end;
+validate_compile_opt(Opt, Defs, Unreachable, File, Line) when is_list(Opt) ->
+  validate_compile_opts(Opt, Defs, Unreachable, File, Line);
+validate_compile_opt(Opt, _Defs, _Unreachable, _File, _Line) ->
+  [Opt].
+
+validate_inline(Inline, Defs, Unreachable) ->
+  case lists:keyfind(Inline, 1, Defs) of
+    false -> throw({bad_inline, Inline});
```

Comment:
Let's please not this style. :) We don't use it anywhere in the codebase and I personally think we should avoid non-local returns. If you think we should start using this, then let's have a discussion, but not in this PR.

If you want to do a single pass, then let's do recursion with `{ok, Var} | {error, Inline}` return.

Reply:
I'm pretty sure that style is a result of spending too much time in OTP internals 😅